pyfr/solvers/euler/elements.py

```diff
@@ -49,6 +48,92 @@ def con_to_pri(cons, cfg):
 
         return [rho] + vs + [p]
 
+    @staticmethod
+    def validate_formulation(ctrl):
+        shock_capturing = ctrl.cfg.get('solver', 'shock-capturing', 'none')
+        if ctrl.formulation == 'dual' and shock_capturing == 'entropy-filter':
+            raise ValueError('Entropy filtering not compatible with '
+                             'dual time stepping.')
+
+        ctrlvardt = ctrl.controller_has_variable_dt
+        if ctrlvardt and shock_capturing == 'entropy-filter':
+            raise ValueError('Entropy filtering not compatible with '
+                             'adaptive time stepping.')
+
+    def set_backend(self, *args, **kwargs):
+        super().set_backend(*args, **kwargs)
+
+        # Can elide shock-capturing at p = 0
+        shock_capturing = self.cfg.get('solver', 'shock-capturing', 'none')
+        if shock_capturing == 'entropy-filter' and self.basis.order != 0:
+            '''
```

Comment:
Probably worth moving up and making a comment.

Reply:
Which part?

Reply:
The documentation block (''').